lessons/33-js-array-map-filter/index.js

```diff
@@ -0,0 +1,107 @@
+const arr1 = [1, 2, 3, 4, 5];
+
+//Map the above array into a new array of squared values and
+// store that in a new variable
+const squar_arr = arr1.map(num => num * num);
+
+// Log both arrays to the console to view the difference
+console.log(arr1);
+console.log(squar_arr);
+
+const arr2 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+// Filter the above values into even numbers only and
+// store the result in a new variable
+const even = arr2.filter(num => {
+  if (num % 2 == 0) return num;
+});
+
+// Filter the above values into odd numbers only and
+// store the result in a new variable
+const odd = arr2.filter(num => {
+  if (num % 2 != 0) return num;
+});
```

Comment:
So unlike `.map`, with filter we want to return `true` or `false`, not the item back (in this case `num`). We return true if we want to keep the item, false if we want to remove it.

So the even one would look like this:

```javascript
const even = arr2.filter(num => {
  if (num % 2 == 0) {
    return true;
  } else {
    return false;
  }
});
```

Or even shorter:

```javascript
const even = arr2.filter(num => {
  return num % 2 == 0;
});
```